service/arm_templates/security_groups_setup.json

```diff
@@ -117,18 +106,32 @@
             }
           },
           {
-            "name": "workerSubnetToMasterSubnet",
+            "name": "allowWorkerSubnet",
             "properties": {
               "description": "Allow the worker machines to reach the master machines on any ports.",
               "protocol": "*",
               "sourcePortRange": "*",
               "destinationPortRange": "*",
               "sourceAddressPrefix": "[parameters('workerSubnetCidr')]",
-              "destinationAddressPrefix": "[parameters('masterSubnetCidr')]",
+              "destinationAddressPrefix": "[parameters('virtualNetworkCidr')]",
```

Comment:
Destination can be VMs subnet or Pods subnet (always within virtual network range). This makes sure that pods from workers can access master.

Alternatively we can add additional rule specifically for Pods. But i prefer to keep rules list as slim as possible. Azure says that many rules affect network performance.